feat: add pipe-cd/pipecd

ponkio-o opened the pull request:
[pipe-cd/pipecd](https://github.com/pipe-cd/pipecd): The One CD for All {applications, platforms, operations}

```console
$ aqua g -i pipe-cd/pipecd
```

## How to confirm if this package works well

Reviewers aren't necessarily familiar with this package, so please describe how to confirm if this package works well.
Please confirm if this package works well yourself as much as possible.

Command and output

piped
```console
root@73bc8118b9a8:/workspace# piped
A component that runs inside target environment to execute deployment and report its state.

Usage:
  piped [command]

Available Commands:
  help        Help about any command
  piped       Start running piped.
  version     Print the information of current binary.

Flags:
  -h, --help                               help for piped
      --log-encoding string                The encoding type for logger [json|console|humanize]. (default "humanize")
      --log-level string                   The minimum enabled logging level. (default "info")
      --metrics                            Whether metrics is enabled or not. (default true)
      --profile                            If true enables uploading the profiles to Stackdriver.
      --profile-debug-logging              If true enables logging debug information of profiler.
      --profiler-credentials-file string   The path to the credentials file using while sending profiles to Stackdriver.

Use "piped [command] --help" for more information about a command.
```

pipectl
```console
root@ec26f5b9726d:/workspace# pipectl
The command line tool for PipeCD.

Usage:
  pipectl [command]

Available Commands:
  application  Manage application resources.
  deployment   Manage deployment resources.
  encrypt      Encrypt the plaintext entered in either stdin or the --input-file flag.
  event        Manage event resources.
  help         Help about any command
  init         Generate an application config (app.pipecd.yaml) easily and interactively.
  piped        Manage piped resources.
  plan-preview Show plan preview against the specified commit.
  quickstart   Quick prepare PipeCD control plane in quickstart mode.
  version      Print the information of current binary.

Flags:
  -h, --help                               help for pipectl
      --log-encoding string                The encoding type for logger [json|console|humanize]. (default "humanize")
      --log-level string                   The minimum enabled logging level. (default "info")
      --metrics                            Whether metrics is enabled or not. (default true)
      --profile                            If true enables uploading the profiles to Stackdriver.
      --profile-debug-logging              If true enables logging debug information of profiler.
      --profiler-credentials-file string   The path to the credentials file using while sending profiles to Stackdriver.

Use "pipectl [command] --help" for more information about a command.
```

Reference

- https://pipecd.dev/

ponkio-o committed:
feat: add pipe-cd/pipecd

[pipe-cd/pipecd](https://github.com/pipe-cd/pipecd): The One CD for All {applications, platforms, operations}

ponkio-o committed:
Merge branch 'main' into feat/pipe-cd/pipecd

suzuki-shunsuke commented:
Thank you always!

suzuki-shunsuke commented:
I'm not familiar with PipeCD, but PipeCD has two tools, pipectl and piped.

https://github.com/pipe-cd/pipecd/releases/tag/v0.47.0

So I think we should change the package name to `pipe-cd/pipecd/piped`.

suzuki-shunsuke commented:
The command name should be piped, not pipecd.
So you have to set `files` if the package name is `pipe-cd/pipecd`.

ponkio-o commented:
> So I think we should change the package name to pipe-cd/pipecd/piped.

Sorry! I will correct it.

ponkio-o committed:
remove: remove old files

ponkio-o committed:
added piped

ponkio-o committed:
added pipectl

ponkio-o committed:
Merge branch 'feat/pipe-cd/pipecd' of https://github.com/ponkio-o/aqua-registry into feat/pipe-cd/pipecd

ponkio-o committed:
Merge branch 'main' into feat/pipe-cd/pipecd

ponkio-o commented:
I removed old files ([commit](https://github.com/aquaproj/aqua-registry/pull/21955/commits/e95fb446b9ef7fdc95075a53093472a9b1947eba)) and added `piped` ([commit](https://github.com/aquaproj/aqua-registry/pull/21955/commits/fcb2b77a003febf5659a9d2721c62662880c12c7)) and `pipectl` ([commit](https://github.com/aquaproj/aqua-registry/pull/21955/commits/1df88c3a42520056f4380a471383027811bdf25c)).

suzuki-shunsuke committed:
fix(pipe-cd/pipecd): fix description

suzuki-shunsuke commented:
nits. updated the description based on help messages.

```console
$ pipectl --help
The command line tool for PipeCD.
```

```console
$ piped --help
A component that runs inside target environment to execute deployment and report its state.
```

suzuki-shunsuke committed:
refactor(pipe-cd/pipecd): remove files

suzuki-shunsuke commented:
v4.163.0 is out 🎉
https://github.com/aquaproj/aqua-registry/releases/tag/v4.163.0
